src/ssh/azext_ssh/bastion_utils.py

```diff
@@ -0,0 +1,355 @@
+def create_bastion(cmd, op_info, vnet_id):
+    try:
+        bastion_creator = Create(cli_ctx=cmd.cli_ctx)
+        bastion_args = {
+            "location": op_info.location,
+            "name": f"{op_info.resource_group_name}-vnet-bastion",
+            "resource_group": op_info.resource_group_name,
+            "sku": "Developer",
+            "virtual_network": {
+                "id": vnet_id
+            },
+            "ip_configurations": [],
+            "scale_units": 2,
+            "tags": {}
+        }
+        result = bastion_creator(command_args=bastion_args)
+         
+        op_info.bastion_name = f"{op_info.resource_group_name}-vnet-bastion"
+    except Exception as e:
+        raise azclierror.ClientRequestError(f"Failed to create bastion information")
+
+#============================================== Get call for current bastion ====================================================#
+def show_bastion(cmd, op_info):
+    from .aaz.latest.network.bastion import Show
+    try:
+        bastion = Show(cli_ctx=cmd.cli_ctx)(command_args={
+            "resource_group": op_info.resource_group_name,
+            "name": op_info.bastion_name 
+        })
+
+        return bastion
+    except Exception:
+        raise azclierror.CLIInternalError("Failed to get bastion information. Please try again later.")
+
+
+
+# ============================= SSH to Bastion Host Logic ============================= #
+def _get_host(username, ip):
+    return username + "@" + ip
+
+
+def ssh_bastion_host(cmd, op_info):
+
+    bastion = show_bastion(cmd, op_info)
+
+    if bastion['sku']['name'] not in [BastionSku.Developer.value, BastionSku.QuickConnect.value]:
+        raise azclierror.InvalidArgumentValueError("SSH to Bastion host is only supported for Developer and QuickConnect Skus.")
+
+    if op_info.port != None:
+        port = op_info.port
+        if port != 22:
+            raise azclierror.InvalidArgumentValueError("Custom Ports are not allowed for the Bastion Developer Sku. Please try again with port 22`.")
+    port =22 
+
+    target_resource_id = op_info.resource_id
+    
+    bastion_endpoint = _get_data_pod(cmd, port, target_resource_id, bastion)
+    tunnel_server = _get_tunnel(cmd, bastion, bastion_endpoint, target_resource_id, port)
+    
+    t = threading.Thread(target=_start_tunnel, args=(tunnel_server,))
+    t.daemon = True
+    t.start()
+
+    command = [ ssh_utils.get_ssh_client_path(), _get_host(op_info.local_user, "localhost")]
+    command = command + op_info.build_args()
+    command = command + ["-p", str(tunnel_server.local_port)]
+    command = command + ["-o", "StrictHostKeyChecking=no", "-o", "UserKnownHostsFile=/dev/null"]
+    command = command + ["-o", "LogLevel=Error"]
+    logger.debug("Running ssh command %s", " ".join(command))
+
+    try:
+        subprocess.call(command, shell=platform.system() == "Windows")
```

Comment:
Can we use the start_ssh_connection function to call SSH? https://github.com/Azure/azure-cli-extensions/blob/940812032876d3f893c6f9a39842fe84bf9ce658/src/ssh/azext_ssh/ssh_utils.py#L26

Here is why:
That function puts together the SSH command, handles cleaning up the credentials once the connection succeeds and handles reading the SSH logs to identify common errors. You don't want to have to re-write all that logic. 

Here is what I'm thinking: you could add the bastion specific SSH parameters to op_info.ssh_args so that start_ssh_connection includes them in the ssh command. You can alter the op_info.get_host() to handle bastion scenarios. Then you call start_ssh_connection and let that handle the ssh connection.